tools/accuracy_checker/accuracy_checker/metrics/detection.py

```diff
@@ -646,42 +664,40 @@ def configure(self):
         self.overlap = self.get_value_from_config('overlap')
         self.relative_size = self.get_value_from_config('relative_size')
 
-    def submit_all(self, annotations, predictions):
-        return self.evaluate(annotations, predictions)
-
     def evaluate(self, annotations, predictions):
         true_positive = 0
         false_positive = 0
 
         for (annotation, prediction) in zip(annotations, predictions):
-            gt_face = [
-                annotation.x_mins[0],
-                annotation.y_mins[0],
-                annotation.x_maxs[0],
-                annotation.y_maxs[0]
-            ]
-            found = False
-            for i in range(prediction.scores.size):
-                dt_face = [
-                    prediction.x_mins[i],
-                    prediction.y_mins[i],
-                    prediction.x_maxs[i],
-                    prediction.y_maxs[i]
+            for gt_idx in range(annotation.x_mins.size):
+                gt_face = [
+                    annotation.x_mins[gt_idx],
+                    annotation.y_mins[gt_idx],
+                    annotation.x_maxs[gt_idx],
+                    annotation.y_maxs[gt_idx]
                 ]
-                iou = self.calc_iou(gt_face, dt_face)
-                if iou:
-                    intersect_area, dt_area, gt_area = iou
-                    if intersect_area / dt_area < self.overlap:
-                        continue
-                    if dt_area / gt_area >= self.relative_size:
-                        found = True
-                        break
-            if found:
-                true_positive += 1
-            else:
-                false_positive += 1
+                found = False
+                for i in range(prediction.scores.size):
+                    dt_face = [
+                        prediction.x_mins[i],
+                        prediction.y_mins[i],
+                        prediction.x_maxs[i],
+                        prediction.y_maxs[i]
+                    ]
+                    iou = self.calc_iou(gt_face, dt_face)
```

Comment:
above you move iou calculation to the function, but inside the class use method.
Is there is reasons for it?
Please stay one calc_iou realization